go/test/endtoend/tabletmanager/tablet_health_test.go

```diff
@@ -244,55 +246,205 @@ func TestHealthCheckDrainedStateDoesNotShutdownQueryService(t *testing.T) {
-	timeout := time.Now().Add(10 * time.Second)
-	for time.Now().Before(timeout) {
-		if tablet.VttabletProcess.WaitForStatus(status) {
-			return nil
-		}
-		time.Sleep(300 * time.Millisecond)
+func TestIgnoreHealthError(t *testing.T) {
+	// This test verify the tablet health by Ignoring the error
+	// For this case we need a healthy tablet in a shard without any master.
+	// When we try to make a connection to such tablet we get "no slave status" error.
+	// We will then ignore this error and verify if the status report the tablet as Healthy.
+
+	// Create a new shard
+	newShard := &cluster.Shard{
+		Name: "1",
 	}
-	return fmt.Errorf("Tablet status is not %s ", status)
+
+	// Start mysql process
+	tablet := clusterInstance.GetVttabletInstance("replica", 0, "")
+	tablet.MysqlctlProcess = *cluster.MysqlCtlProcessInstance(tablet.TabletUID, tablet.MySQLPort, clusterInstance.TmpDirectory)
+	err := tablet.MysqlctlProcess.Start()
+
+	// start vttablet process
+	tablet.VttabletProcess = cluster.VttabletProcessInstance(tablet.HTTPPort,
+		tablet.GrpcPort,
+		tablet.TabletUID,
+		clusterInstance.Cell,
+		newShard.Name,
+		clusterInstance.Keyspaces[0].Name,
+		clusterInstance.VtctldProcess.Port,
+		tablet.Type,
+		clusterInstance.TopoProcess.Port,
+		clusterInstance.Hostname,
+		clusterInstance.TmpDirectory,
+		clusterInstance.VtTabletExtraArgs,
+		clusterInstance.EnableSemiSync)
+	tablet.Alias = tablet.VttabletProcess.TabletPath
+	newShard.Vttablets = append(newShard.Vttablets, tablet)
+
+	clusterInstance.Keyspaces[0].Shards = append(clusterInstance.Keyspaces[0].Shards, *newShard)
+
+	// Init Tablet
+	err = clusterInstance.VtctlclientProcess.InitTablet(tablet, cell, keyspaceName, hostname, newShard.Name)
+	assert.Nil(t, err)
+
+	// create database
+	err = tablet.VttabletProcess.CreateDB(keyspaceName)
+	assert.Nil(t, err)
+
+	// Start Vttablet, it should be NOT_SERVING as there is no master
+	err = clusterInstance.StartVttablet(tablet, "NOT_SERVING", false, cell, keyspaceName, hostname, newShard.Name)
+	assert.Nil(t, err)
+
+	// Force it healthy.
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("IgnoreHealthError", tablet.Alias, ".*no slave status.*")
+	assert.Nil(t, err)
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("RunHealthCheck", tablet.Alias)
+	assert.Nil(t, err)
+	err = tablet.VttabletProcess.WaitForTabletType("SERVING")
+	assert.Nil(t, err)
+	checkHealth(t, tablet.HTTPPort, false)
+
+	// Turn off the force-healthy.
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("IgnoreHealthError", tablet.Alias, "")
+	assert.Nil(t, err)
+	err = clusterInstance.VtctlclientProcess.ExecuteCommand("RunHealthCheck", tablet.Alias)
+	assert.Nil(t, err)
+	err = tablet.VttabletProcess.WaitForTabletType("NOT_SERVING")
+	assert.Nil(t, err)
+	checkHealth(t, tablet.HTTPPort, true)
+
+	// Tear down custom processes
+	killTablets(t, tablet)
 }
 
-func TestIgnoreHealthError(t *testing.T) {
+func TestNoMysqlHealthCheck(t *testing.T) {
```

Comment:
Does this mean that we are not testing the IgnoreHealthError functionality? Is there a different test case that covers it?

Reply:
That's just github's view. They usually mess up if the same signature method is added up/down. 
The `TestIgnoreHealthError` is still present at line 249.

Reply:
Ah ok